### ISSUE TITLE
docs: Alternator compatibility page become to long

### ISSUE THREAD

**Issue by tzach** [scylladb/scylladb#19822]
I would like to report an issue on page https://opensource.docs.scylladb.com/branch-6.0/alternator/compatibility

### Problem
Two problems with this page's readability:
- The URL, "compatibility", does not match the title "ScyllaDB Alternator for DynamoDB users" or the content
- The page becomes too long

###  Suggest a fix

- Update the URL (with a proper redirection)  to something like "using-alternator" to match https://opensource.docs.scylladb.com/stable/using-scylla/index.html
- Break the page to multiple pages under "using-alternator" path

**Comment by nyh** [comment]
Tzach, I wouldn't call that page "using Alternator", because the whole point was using Alternator **for DynamoDB users**, i.e., what existing DynamoDB users need to know to switch to Alternator. In other words, Alternator's **compatibility** with DynamoDB, and what to do when it's not compatible. This is where the name "compatibility" came from.

I don't agree that the page is "too long" and against just splitting it: The goal of this page is to have a page where someone who is already using DynamoDB can read - from the first to the last line - and see which features their application uses that might be problematic on Alternator. I don't want them to miss some of these features just because it's on a separate page.

The best (if not only) way I see to make this page shorter is to make Alternator even more compatible with DynamoDB. About a third of the page is "Unimplemented API features", if we had fewer of those the page would have been shorter.

Another thing we can do is when we have a non-compatible approach to doing something (e.g., authorization), create a whole page just on that, and leave only one paragraph in compatibility.md. Maybe this is what you had in mind when you suggested breaking up compatibility.md. But I would do this only for a couple of topics (like authorization) and not break the page up to many small pages.

**Comment by tzach** [comment]
Most users would expect a simple comparison table to understand compatibility. Some need more details, but not all, and each user might be interested in a different topic.

Setting permission, for example, is part of "Using Alternator", the same way that CQL is part of using Scylla, even though Scylla is Cassandra compatible.

The Doc format needs to support long pages better.
It's easier to read and navigate between pages, and no links are on the same page. In most cases, no one will read the entire page; instead, they will look for specific items. A search engine pointing directly to a (short) page is more accessible.

**Comment by nyh** [comment]
> Setting permission, for example, is part of "Using Alternator", the same way that CQL is part of using Scylla, even though Scylla is Cassandra compatible.

I agree that for the authorization part (which I expend even more in a new PR) it might make sense to have a whole new page just about it, and in the general compatibility page leave just one paragraph saying it's different. That's a valid point for my RBAC PR.
But I think there aren't many other topics on the current compatibility.md which make sense to spin off to a separate page. You don't want to create a new page with 5 lines. Can you give another example of a long section you would split to a different page?

> The Doc format needs to support long pages better. It's easier to read and navigate between pages,

But in this case, the goal of this page (at least my goal when I wrote it) was to have the user read the **whole** page and think for each one of the 17 or whatever features it lists - whether your application is using this feature or not. Users shouldn't have to navigate this page or seach in it - they should read it top to bottom. At least that was my opinion.

**Comment by nyh** [comment]
> I agree that for the authorization part (which I expend even more in a new PR) it might make sense to have a whole new page just about it, and in the general compatibility page leave just one paragraph saying it's different. That's a valid point for my RBAC PR.

In the review of that PR, @avikivity also made this point - https://github.com/scylladb/scylladb/pull/19762/commits/da4ee8bbec3d7cd455a23dbd4fa75017b040980d#r1703369777 

Indeed, we can have a whole separate page on authentication and authorization in Alternator, which acknowledges that our RBAC is not just a small incompatibility with DynamoDB - it's an entirely different feature that deserves its own documentation. Then compatibility.md will just mention this as a difference, and refer to the other page (e.g. auth.md) for details.
